server/src/main/java/org/opensearch/action/admin/cluster/decommission/awareness/put/TransportPutDecommissionAction.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.action.admin.cluster.decommission.awareness.put;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.action.ActionListener;
+import org.opensearch.action.support.ActionFilters;
+import org.opensearch.action.support.clustermanager.TransportClusterManagerNodeAction;
+import org.opensearch.cluster.ClusterState;
+import org.opensearch.cluster.block.ClusterBlockException;
+import org.opensearch.cluster.block.ClusterBlockLevel;
+import org.opensearch.cluster.metadata.IndexNameExpressionResolver;
+import org.opensearch.cluster.service.ClusterService;
+import org.opensearch.common.inject.Inject;
+import org.opensearch.common.io.stream.StreamInput;
+import org.opensearch.threadpool.ThreadPool;
+import org.opensearch.transport.TransportService;
+
+import java.io.IOException;
+
+public class TransportPutDecommissionAction extends TransportClusterManagerNodeAction<PutDecommissionRequest, PutDecommissionResponse> {
```

Comment:
what is expectation from the API, it will synchronously process the decommissioning or it will trigger it in the background?

Reply:
The API will return the acknowledged response to the user as soon as the metadata is registered with `DECOMMISSION_INIT` status. And rest of the operation will be performed in the background. The user need to trigger the GET call to get the current status of decommissioning.

Implementation for above is part of this PR - #4084